common/views/components/ApiToolbar/ApiToolbar.tsx

```diff
@@ -37,26 +38,78 @@ const includes = [
  * from the catalogue.
  *
  */
-async function createTzitzitLink(imageId: string): Promise<Prop> {
-  const apiUrl = `https://api.wellcomecollection.org/catalogue/v2/images/${imageId}?include=source.contributors`;
-  const image: Image = await fetch(apiUrl).then(res => res.json());
+
+function setTzitzitParams(data: {
+  title: string;
+  sourceLink: string;
+  licence: string;
+  author?: string;
+}): string {
+  // If license is INC, return as the tool should not be made available
+  if (data.licence === 'INC') return '';
 
   const params = new URLSearchParams();
-  params.set('title', image.source.title);
+  params.set('title', data.title);
   params.set('sourceName', 'Wellcome Collection');
-  params.set('sourceLink', window.location.toString());
-  params.set('licence', image.locations[0].license.id.toUpperCase());
+  params.set('sourceLink', data.sourceLink);
+  if (data.licence) params.set('licence', data.licence);
+  if (data.author) params.set('author', data.author);
+
+  return params.toString();
+}
 
+async function createTzitzitImageLink(imageId: string): Promise<Prop> {
+  const apiUrl = `https://api.wellcomecollection.org/catalogue/v2/images/${imageId}?include=source.contributors`;
+  const image: Image = await fetch(apiUrl).then(res => res.json());
   const contributors = image.source.contributors;
 
-  if (contributors && contributors.length > 0) {
-    params.set('author', contributors[0].agent.label);
-  }
+  const params = setTzitzitParams({
+    title: image.source.title,
+    sourceLink: window.location.toString(),
+    licence: image.locations[0].license.id.toUpperCase(),
+    author:
+      contributors && contributors.length > 0
+        ? contributors[0].agent.label
+        : '',
```

Comment:
Minor: since we're doing this for both Miro images and METS, maybe let's pass in the list of contributors and run this label-based logic there?

Reply:
That's what I did at first, but the typing of Contributors confused me slightly (I know it exists in other parts of the project), could we jump on a call to figure it out after standups?

Reply:
Yup, sure.